core/src/banking_stage/consume_worker.rs

```diff
@@ -122,6 +148,460 @@ fn try_drain_iter<T>(work: T, receiver: &Receiver<T>) -> impl Iterator<Item = T>
     std::iter::once(work).chain(receiver.try_iter())
 }
 
+/// Metrics tracking number of packets processed by the consume worker.
+/// These are atomic, and intended to be reported by the scheduling thread
+/// since the consume worker thread is sleeping unless there is work to be
+/// done.
+pub(crate) struct ConsumeWorkerMetrics {
+    id: u32,
+    interval: AtomicInterval,
+    latch: AtomicBool,
+
+    count_metrics: ConsumeWorkerCountMetrics,
+    error_metrics: ConsumeWorkerTransactionErrorMetrics,
+    timing_metrics: ConsumeWorkerTimingMetrics,
+}
+
+impl ConsumeWorkerMetrics {
+    /// Report and reset metrics iff the interval has elapsed and the worker did some work.
+    pub fn maybe_report_and_reset(&self) {
+        const REPORT_INTERVAL_MS: u64 = 1000;
+        if self.interval.should_update(REPORT_INTERVAL_MS)
```

Comment:
Is it not useful or meaningful to aggregate these metrics per slot? Banking_stage reports bunch of `leader_slot...` metrics

Reply:
probably OK for now to report by interval. I'm thinking it'd be necessary to look at per-block data in order to monitor priority quality (eg how priority if respected) and how many priority fee are collected etc

Reply:
I want to add the fee collection metrics on the QoS stats; that way it's reported by both the old and new banking stage approaches